src/OrchardCore.Modules/OrchardCore.Tenants/Workflows/Activities/SetupTenantTask.cs

```diff
@@ -120,7 +119,7 @@ public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecuti
                 }
 
                 ShellSettingsManager.SaveSettings(shellSettings);
```

Comment:
idem, you can remove this line

Then, if this task can be run separately, not just after a create tenant task, maybe after having done `ShellHost.TryGetSettings()`, maybe good to do `ShellSettingsManager.LoadSettings(settings.Name)`, i will let you know.

Here i asked me that, if for a given tenant name, no settings are found (or if no tenant name provided), why not just returning `Failed`. Will see when i will ask other questions, only simple questions not so much ;)